### ISSUE TITLE
onDismissed is not called.

### ISSUE THREAD

**Issue by tmztr** [salim-lachdhaf/dropdown_search#693]
Simply with this code, the "onDismissed" does not execute after closing the popup menu.

```
DropdownSearch<String>.multiSelection(
                items: (f, cs) => ["Item 1", 'Item 2', 'Item 3', 'Item 4'],
                popupProps: PopupPropsMultiSelection.menu(
                  onDismissed: () {
                    // This does not fire.
                    print("Dismissed");
                  },
                  disabledItemFn: (item) => item == 'Item 3',
                  fit: FlexFit.loose,
                ),
              ),
            )
```

**Comment by salim-lachdhaf** [comment]
Bug not reproducible in v6.0.1 on Android.
Could you indique in which platform and version you get this issue ?

**Comment by tmztr** [comment]
This is reproducible on web with chrome.

**Comment by salim-lachdhaf** [comment]
tested in Google chrome => no issue.
![image](https://github.com/user-attachments/assets/eaad88bf-411e-45ba-bfd2-3b114b662e9c)

Could you put a video or a gif to reproducible issue ?

**Comment by tmztr** [comment]
Using v6.0.1: onDismissed is not triggered.
[v6.0.1.webm](https://github.com/user-attachments/assets/aa2974ec-c9df-432a-b585-1622ebf1261b)

Using v6.0.0: onDismissed is actually triggered.
[v6.0.0.webm](https://github.com/user-attachments/assets/e661713a-4307-45c2-95ba-7c3930e9eae1)

Here is my setup:
Operating System: Ubuntu 22.04.5 LTS              
Kernel: Linux 6.8.0-45-generic
Flutter 3.24.3
Dart 3.5.3

Performed on Google Chrome 129.0.6668.89.

**Comment by malwinder-s** [comment]
this issue is coming on the Android simulator too API Vanilla Ice Cream

**Comment by lambej** [comment]
The onDismissed is triggering on Android with v6.0.4-pre

**Comment by AlessandroDeStefano** [comment]
I confirm I have the same problem with v6.0.1 in web chrome but not with the v6.0.0

**Comment by salim-lachdhaf** [comment]
@AlessandroDeStefano  onDismissed called inside onDispose of the popup.  It should be called every time the popup is closed.
The problem it's is working in same devices and not in others.

@AlessandroDeStefano @malwinder-s @tmztr 
could you please confirm that you see `destroyed` in console in this example after running it in device with the issue ?:

``` 
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Two Empty Pages',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: FirstPage(),
    );
  }
}

class FirstPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('First Page'),
      ),
      body: Center(
        child: ElevatedButton(
          child: Text('Go to Second Page'),
          onPressed: () {
            Navigator.push(
              context,
              MaterialPageRoute(builder: (context) => SecondPage()),
            );
          },
        ),
      ),
    );
  }
}

class SecondPage extends StatefulWidget {
  @override
  State<StatefulWidget> createState() => SecondPageState();
}

class SecondPageState extends State<SecondPage> {
  @override
  void dispose() {
    print('destroyed');
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Second Page'),
      ),
      body: Center(
        child: ElevatedButton(
          child: Text('Finish'),
          onPressed: () => Navigator.of(context).pop(),
        ),
      ),
    );
  }
}

```

**Comment by CoreySafetyIndicators** [comment]
I am seeing the same issue when using Chrome with v6.0.1.  When clicking outside of the menu, onDismissed is not called even though the menu goes away.  I tried reverting back to 6.0.0 to test it but was having difficulties using that version.

**Comment by salim-lachdhaf** [comment]
@CoreySafetyIndicators 
could you please confirm that you see destroyed in console in this example after running it in device with the issue ?:
```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Two Empty Pages',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: FirstPage(),
    );
  }
}

class FirstPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('First Page'),
      ),
      body: Center(
        child: ElevatedButton(
          child: Text('Go to Second Page'),
          onPressed: () {
            Navigator.push(
              context,
              MaterialPageRoute(builder: (context) => SecondPage()),
            );
          },
        ),
      ),
    );
  }
}

class SecondPage extends StatefulWidget {
  @override
  State<StatefulWidget> createState() => SecondPageState();
}

class SecondPageState extends State<SecondPage> {
  @override
  void dispose() {
    print('destroyed');
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Second Page'),
      ),
      body: Center(
        child: ElevatedButton(
          child: Text('Finish'),
          onPressed: () => Navigator.of(context).pop(),
        ),
      ),
    );
  }
}
```

**Comment by salim-lachdhaf** [comment]
I think it's related to this issue https://github.com/flutter/flutter/issues/40940

In all cases i think i have a workaround to fix it in the next version

**Comment by CoreySafetyIndicators** [comment]
@salim-lachdhaf Sorry for the delay.  I ran your example and I do see destroy printed in the console.  But it sounds like you may have located the issue.  Thank you for looking at this.

**Comment by AlessandroDeStefano** [comment]
@salim-lachdhaf I'm sorry for the very delayed response. I do see destroy printed in the console.
You answered me:
> onDismissed called inside onDispose of the popup.

But I can't find this call you mention in the codebase of the package for the latest version. Can you point me to the place in the code where the `onDismissed` callback is called?

**Comment by salim-lachdhaf** [comment]
this issue is solved in v7.0.0-pre (will be release today). re open it if not the case.
